modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationHolder.java

```diff
@@ -0,0 +1,246 @@
+     */
+    public Optional<String> getOptProperty(String key) {
+        return Optional.ofNullable(userConfig.getProperty(key));
+    }
+
+    /**
+     * Returns the property value as a boolean associated with the given {@code key} from the user configuration.
+     */
+    public boolean getPropertyAsBoolean(String key) {
+        return Boolean.parseBoolean(getProperty(key, "false"));
+    }
+
+    /**
+     * Returns the property value as a int associated with the given {@code key} from the user configuration.
+     *
+     * @throws NumberFormatException – if the string does not contain a parsable integer
+     */
+    public int getPropertyAsInteger(String key, int defaultValue) {
+        return Integer.parseInt(getProperty(key, String.valueOf(defaultValue)));
+    }
+
+    /**
+     * Returns the property value as a {@link Path} associated with the given {@code key} from the user configuration.
+     *
+     * @throws InvalidPathException – if the path string cannot be converted to a Path
+     */
+    public Path getPropertyAsPath(String key) {
+        return Path.of(getProperty(key));
+    }
+
+    /**
+     * Returns the raw property value associated with the given {@code key} from the user configuration.
+     */
+    public String getRawProperty(String key) {
+        return userConfig.getRawProperty(key);
+    }
+
+    /**
+     * Returns the user configuration key set.
+     *
+     * @see Properties#keySet()
+     */
+    @SuppressWarnings("unchecked")
+    public Set<String> keySet() {
+        return (Set<String>) (Set<?>) userConfig.keySet();
+    }
+
+    /**
+     * Returns the user configuration property names.
+     *
+     * @see Properties#stringPropertyNames()
+     */
+    public Set<String> stringPropertyNames() {
+        return userConfig.stringPropertyNames();
+    }
+
+    /**
+     * Puts the given property into default configuration.
+     */
+    public String putDefault(String key, String value) {
+        return (String) defaultConfig.put(key, value);
+    }
+
+    /**
+     * Puts all given {@code properties} into default configuration.
+     */
+    public void putDefaultAll(Properties properties) {
+        defaultConfig.putAll(properties);
+    }
+
+    /**
+     * Puts the given property into user configuration.
+     */
+    public String put(String key, String value) {
+        return (String) userConfig.put(key, value);
+    }
+
+    /**
+     * Puts all given {@code properties} into user configuration.
+     */
+    public void putAll(Properties properties) {
+        userConfig.putAll(properties);
+    }
+
+    protected void clear() {
+        defaultConfig.clear();
+        userConfig.clear();
+    }
+
+    protected void setConfigurable(boolean configurable) {
+        this.configurable = configurable;
+        if (!configurable) {
+            // clear the configuration - probably an error case
```

Comment:
you can use the above clear() method.
where is used the setConfigurable() method and why clearing the config when it's false?
maybe is it a remnant of the "static server" with read-only configuration?

Reply:
I didn't find any usage of `ConfigurationGenerator#isConfigurable`, I can't tell for what it was used. Maybe it was for the static server indeed.

At this point of the changes, I was clearing the config when it was false because it was the behavior the generator was doing when this flag was false. I will try to clean this change from the intermediate commit as it doesn't make sense because the whole thing was replaced by has the config been loaded.